src/test/groovy/org/prebid/server/functional/testcontainers/PbsServiceFactory.groovy

```diff
@@ -22,13 +22,13 @@ class PbsServiceFactory {
     }
 
     PrebidServerService getService(Map<String, String> config) {
```

Comment:
This method is not thread safe, it contains so-called check-then-act mistakes. Checking state of collection and updating it should be a single atomic operation. Consider using locks or synchronized blocks. Also, please check for another methods that are called on the same object from different threads.